xbmc/network/upnp/UPnPInternal.cpp

```diff
@@ -1218,5 +1219,44 @@ std::shared_ptr<CFileItem> GetFileItem(const NPT_String& uri, const NPT_String&
     return item;
 }
 
+NPT_String EncodeObjectId(const std::string& id)
+{
+  if (id.empty())
+  {
+    CLog::LogF(LOGWARNING, "Failed to encode object id, provided object id is empty");
+    return {};
+  }
+  // we use integers in some special cases like virtualpath://upnproot or whenever clients with
+  // quirks expect to receive integer ids for some parent containers
+  if (StringUtils::IsInteger(id))
+  {
+    return id.c_str();
+  }
+
+  return Base64::Encode(id).c_str();
+}
+
+NPT_String DecodeObjectId(const std::string& id)
+{
+  if (id.empty())
+  {
+    CLog::LogF(LOGWARNING, "Failed to decode object id, provided object id is empty");
+    return {};
+  }
+  // we use integers in some special cases like virtualpath://upnproot or whenever clients with
+  // quirks expect to receive integer ids for some parent containers
+  if (StringUtils::IsInteger(id))
```

Comment:
Is it possible that the base64 encoded string is an integer? That might lead to misinterpretations here.

Reply:
Nice point. I guess it is possible but highly unlikely due to the way base64 works (multiple conversions involved) and also what we "expect" to have as decoded values. 
Maybe this can be made safer by defining some whitelist of allowed values.